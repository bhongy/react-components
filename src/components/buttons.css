.base {
  border: none;
  min-width: 5rem;
  padding: 0.5rem 1rem;
  cursor: pointer;
  text-transform: uppercase;

  &:focus {
    outline: none;
  }
}

.raised {
  composes: base;
  border-radius: 0.25rem;
  transition-properties: background-color, box-shadow;
  transition-duration: $duration-fast;
}

.raised__active {
  composes: raised;
  background-color: #303f9f;
  color: white;
  box-shadow: 0 3px 6px color(black alpha(0.15));
}

.raised__disabled {
  composes: raised;
  background-color: #ddd;
  color: white;
  cursor: default;
  box-shadow: 0 3px 6px color(black alpha(0));
}

.flat {
  composes: base;
  color: #303f9f;
  background-color: transparent;
  transition: box-shadow $duration-fast;

  &:hover {
    box-shadow: 0 3px 6px color(black alpha(0.05));
  }
}

.flat__disabled {
  composes: flat;
  opacity: $opacity-light;
  cursor: default;

  &:hover {
    box-shadow: none;
  }
}
