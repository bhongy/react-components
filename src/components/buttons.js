/* eslint-disable import/prefer-default-export */

import React from 'react';
import cx from 'classnames';
import s from './buttons.css';

export const RaisedButton = ({ className, disabled, ...otherProps }) => {
  const baseClassName = disabled ? s.raised__disabled : s.raised__active;
  return (
    <button
      {...otherProps}
      disabled={disabled}
      className={cx(baseClassName, className)}
    />
  );
};

export const FlatButton = ({ className, disabled, ...otherProps }) => {
  const baseClassName = disabled ? s.flat__disabled : s.flat;
  return (
    <button
      {...otherProps}
      disabled={disabled}
      className={cx(baseClassName, className)}
    />
  );
};
