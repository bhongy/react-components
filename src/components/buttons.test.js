import React from 'react';
import { shallow } from 'enzyme';
import { RaisedButton } from './buttons';

describe('Raised Button', () => {
  it('renders button', () => {
    expect(shallow(<RaisedButton />).type()).toBe('button');
  });
});
