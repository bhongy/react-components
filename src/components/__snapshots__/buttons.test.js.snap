// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flat Button renders default button 1`] = `
<button
  className="flat"
/>
`;

exports[`Flat Button renders disabled button 1`] = `
<button
  className="flat__disabled"
  disabled={true}
/>
`;

exports[`Raised Button renders default button 1`] = `
<button
  className="raised__active"
/>
`;

exports[`Raised Button renders disabled button 1`] = `
<button
  className="raised__disabled"
  disabled={true}
/>
`;
